src/components/widgets/accounts-cash-projection/accounts-cash-projection.directive.coffee

```diff
@@ -173,6 +184,14 @@ module.controller('WidgetAccountsCashProjectionCtrl', ($scope, $q, $filter, $tim
       continue if s.userOptions.linkedTo != series.name
       if series.visible then s.hide() else s.show()
 
+  addCustomHighChartOptions = () ->
+    highChartsFactory = new HighchartsFactory($scope.chartId(), w.content.chart.series, _highChartOptions)
+    highChartsFactory.addCustomLegend(legendFormatter)
+    highChartsFactory.addSeriesEvent('click', onClickBar)
+    highChartsFactory.addSeriesEvent('legendItemClick', onClickLegend)
+    highChartsFactory.addXAxisOptions(_highChartOptions.withZooming)
```

Comment:
I like the setters, means widgets can really configure the charts with a nice api. 

Thought It feels weird giving the factory the full `_highChartOptions` on init, then accessing one of the attributes for the `addXAxisOptions` method. I wouldn't shy away from using this syntax:

```coffeescript
highChartsFactory.addXAxisOptions({
    defaults: w.metadata.xAxis
    callback: onZoom
})
```

It means I you dont have to go scrolling up and down the file looking for the options hash, keeping things together. I think we should move the whole options to this if possible, see my comment at the `_highChartOptions` declarations.

Reply:
Good point. What exactly do you mean by "we should move the whole options to this if possible."

Reply:
@iseessel never-mind about that comment actually